dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/MergeUser/ChooseTrn.cshtml.cs

```diff
@@ -0,0 +1,70 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer.Pages.Admin.MergeUser;
+
+public class ChooseTrn : PageModel
+{
+    public new User? User { get; set; }
+    public User? UserToMerge { get; set; }
+
+    private readonly TeacherIdentityServerDbContext _dbContext;
+
+    public ChooseTrn(TeacherIdentityServerDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    [BindProperty]
+    [Display(Name = "Which TRN do you want to keep?")]
+    [Required(ErrorMessage = "Select the TRN you want to keep")]
+    public string? Trn { get; set; }
+
+    [FromRoute]
+    public Guid UserId { get; set; }
+
+    [FromRoute]
+    public Guid UserIdToMerge { get; set; }
+
+
+    public void OnGet()
+    {
+    }
+
+    public IActionResult OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        HttpContext.Session.SetString(Confirm.MergeTrnKey, Trn!);
```

Comment:
Ok for now but would be good to ultimately get it into the URL (a query param, perhaps?)

Reply:
I figured that if we allow any other choices e.g. name/email then it would be odd to continue to pass around the chosen TRN in the query param, and that it ought to be stored somewhere server side